packages/grafana-data/src/types/datasource.ts

```diff
@@ -492,6 +492,8 @@ export interface DataQueryRequest<TQuery extends DataQuery = DataQuery> {
   timeInfo?: string; // The query time description (blue text in the upper right)
   panelId?: number;
   dashboardId?: number;
+  dashboardUid?: string;
+  userCanEditDashboard?: boolean;
```

Comment:
This seems wrong to pass this flag via the `DataQueryRequest`. I guess this information comes from boot config or  in some other way that can be used in `getUrlForRequest`?

Reply:
To clarify, you're referring to `userCanEditDashboard` correct?

We're not sure if we can get those permissions from the boot config because the FGAC dashboard permissions are so specific. Are those FGAC permissions loaded into the boot config?

Reply:
In general feels like this validation should happen in the backend. So is it really needed client side?

Reply:
That seems reasonable. I'll take a look and see if we can figure this out from the current auth token + FGAC in the backend.

Reply:
wonder if this should instead be replaced with: 

`validateQuery?: ValidateQueryInfo`

That we only set when !meta.canEdit or it's a public dashboard. 

ValidateQueryInfo is an interface with panelId, dashboardUid and maybe other things in the future (panelUid for a library panel, and maybe some future queryUId)